eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -279,28 +281,48 @@ class ReplicationEndpoint(
     else if (active.compareAndSet(false, true)) {
       import system.dispatcher
 
-      val promise = Promise[Unit]()
       val recovery = new Recovery(this)
 
       def recoveryFailure[U](partialUpdate: Boolean): PartialFunction[Throwable, Future[U]] = {
         case t => Future.failed(new RecoveryException(t, partialUpdate))
       }
 
-      val recoveryOperation = for {
-        infos <- recovery.readEndpointInfos.recoverWith(recoveryFailure(partialUpdate = false))
-        clocks <- recovery.readEventLogClocks.recoverWith(recoveryFailure(partialUpdate = false))
-        links = recovery.recoveryLinks(infos, clocks)
-        _ <- recovery.deleteSnapshots(links).recoverWith(recoveryFailure(partialUpdate = true))
-      } yield acceptor ! Recover(links, promise)
+      for {
+        localEndpointInfo <- recovery.readEndpointInfo.recoverWith(recoveryFailure(partialUpdate = false))
+        _ = logLocalState(localEndpointInfo)
+        remoteEndpointInfos <- recovery.synchronizeReplicationProgressesWithRemote(localEndpointInfo).recoverWith(recoveryFailure(partialUpdate = false))
+        _ = logRemoteState(remoteEndpointInfos)
+        unfilteredLinks = recovery.recoveryLinks(remoteEndpointInfos, localEndpointInfo, filtered = false)
+        _ = logLinksToBeRecovered(unfilteredLinks, "unfiltered")
+        _ <- recovery.recoverLinks(unfilteredLinks).recoverWith(recoveryFailure(partialUpdate = true))
+        filteredLinks = recovery.recoveryLinks(remoteEndpointInfos, localEndpointInfo, filtered = true)
```

Comment:
The API doc actually states: `After successful recovery the endpoint is automatically activated.` I will add a comment why unfiltered connections have to be recovered first.